livekit/sip_validation.go

```diff
@@ -0,0 +1,359 @@
+// Copyright 2023 LiveKit, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package livekit
+
+import (
+	"errors"
+	fmt "fmt"
+	"strconv"
+	"strings"
+)
+
+// RFC 3261 compliant validation functions for SIP headers and messages
+
+type AllowedCharacters struct {
```

Comment:
This should be private.
```suggestion
type allowedCharacters struct {
```

Reply:
We could always re-export it later. But SIP can use a real parser, so probably not.